### PR TITLE
refactor: Migrate team permission utils to modules

### PR DESCRIPTION
**Pull request by atomrc**
This is only a refactoring PR to prepare for the next feature. 

It doesn't include any behavior change. 

I bootstrapped tests for the Permission handler and refactored the tested method. 

I also resolved a cyclic dependency between `permission` and `role` handlers. We might want to re-split those later on but let's do it properly when we can take a step back on what we need exactly.

### CODE REVIEW

File: src/script/team/TeamPermission.js [hunk 1]
```diff
@@ -17,89 +17,101 @@
  *
  */
 
-window.z = window.z || {};
-window.z.team = z.team || {};
+/**
+ * Enum for different team permissions.
+ * @returns {z.team.TeamPermission.PERMISSION} Enum of team permissions
+ */
+const PERMISSION = {
```
Comment (AndyLnd) [inline src/script/team/TeamPermission.js:25]: Can we sort that by value and not alphabetical, please?

File: src/script/team/TeamPermission.js [hunk 2]
```diff
@@ -17,89 +17,101 @@
...
+
+  return detectedRole || ROLE.INVALID;
+}
+
+function combinePermissions(permissions) {
+  let result = 0;
+  for (const permission of permissions) {
+    result = result | permission;
+  }
+  return result;
```
Comment (AndyLnd) [inline src/script/team/TeamPermission.js]: ```return permissions.reduce((acc, permission) => acc | permission, 0);```

File: test/unit_tests/team/TeamPermissionSpec.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {ROLE, roleFromPermissions} from 'src/script/team/TeamPermission';
+
+const memberPermissionBitmask = 1587;
```
Comment (AndyLnd) [inline test/unit_tests/team/TeamPermissionSpec.js]: Can we please write those in binary notation, so they make more sense as a bitmask?
```suggestion
const memberPermissionBitmask = 0b11000110011;
```

File: test/unit_tests/team/TeamPermissionSpec.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+const ownerPermissionBitmask = 8191;
```
Comment (AndyLnd) [inline test/unit_tests/team/TeamPermissionSpec.js]:
```suggestion
const ownerPermissionBitmask = 0b1111111111111;
```

File: test/unit_tests/team/TeamPermissionSpec.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+const adminPermissionBitmask = 5951;
```
Comment (AndyLnd) [inline test/unit_tests/team/TeamPermissionSpec.js]:
```suggestion
const adminPermissionBitmask = 0b1011100111111;
```

File: src/script/team/TeamPermission.js (hunk 1 above)
Comment (atomrc) [reply src/script/team/TeamPermission.js:25]: I wouldn't say value is a better sorting algo than feature name....
Why do you feel it's better?

Comment (AndyLnd) [reply src/script/team/TeamPermission.js:25]: Maybe it's just my OCD kicking in, but I feel that with bitflags the value is way more important than the name that you chose for it.

**Pull request merged**